internal/dataplane/translator/subtranslator/httproute.go

```diff
@@ -482,11 +480,12 @@ func mergePluginsOfTheSameType(plugins []kong.Plugin) ([]kong.Plugin, error) {
 		return *p.Name // Name is effectively a plugin type.
 	})
 	for pluginName, plugins := range pluginsByName {
+		plugins := plugins
 		// If we produced multiple plugins of the same type, we need to merge their configurations now.
 		if len(plugins) > 1 {
 			mergedPlugin := *plugins[0].DeepCopy()
-			for _, plugin := range plugins {
-				if err := mergo.Merge(&mergedPlugin.Config, plugin.Config); err != nil {
+			for _, plugin := range plugins[1:] {
+				if err := mergo.Merge(&mergedPlugin.Config, plugin.Config, mergo.WithAppendSlice); err != nil {
```

Comment:
I assume there's some slice handling nonsense that necessitates the second instantiation and the partial range, but it's not immediately obvious. Can you add a comment explaining why it's not a basic range loop?

Reply:
Yeah, sure. It's all about not merging a first plugin's config into itself as with `mergo.WithAppendSlice` that would result in duplicate entries in slices from it. Added comments. https://github.com/Kong/kubernetes-ingress-controller/pull/5952/commits/83644e922238902101240b09ccc5dd9baa0d0280